BHoM/SQL_Toolkit
# Upgrade to NetStandard

### Issues addressed by this PR
Fixes #31 - second attempt!


### Test files

1. Install a recent alpha or the 6.1 beta (either works).
2. Delete `SQL_Adapter.dll`, `SQL_Engine.dll`, and `SQL_oM.dll` from your local assemblies
3. Compile in `debug` mode - this should give you the same DLLs as the installer
4. Run test scripts of your choosing - these should run successfully without errors
5. Compile in `ZeroCodeTool` mode - this should give you a new output folder in the `SQL_Toolkit/Build` location
6. Hmm - how to test this bit?!?
7. I'd recommend making a new `.Net6` project, referencing the DLLs and try to pull some data replicating a script used in step 4. Providing the code compiles without compiler warnings (compiler warnings should relate to trying to use `.NetFramework` in `.Net` environments - these should NOT appear with this work) and data can be pulled into a console, this should be sufficient?

[SQL_Toolkit_Test.zip](https://github.com/BHoM/SQL_Toolkit/files/11297572/SQL_Toolkit_Test.zip)

I'm a bit kind - this zip file should help with testing steps 6 onwards - it contains a `.Net6` solution with references to the right DLLs/nugets that would come if the `NetStandard` adapter was included via a nuget (the ultimate end result for this work). You will need to reference the `SQL_Adapter`, `SQL_Engine`, and `SQL_oM` from your local build of `SQL_Toolkit/Build/netstandard2.0` folder - but the other DLLs are already hardcoded for `C:/ProgramData/BHoM` so hopefully minimal set up needed!

Also feel free to not use this - but it's what I made to help debug/run through my own testing instructions so figured I might as well share it! 😄 

### Additional comments
Other work to be included for audit as to how this approach ended up being picked:
PR #32  - original attempt to upgrade
PR #35  - fixing post build for adapter
PR #36  - reverting original attempt
